extensions/smallrye-metrics/spi/src/main/java/io/quarkus/smallrye/metrics/deployment/spi/MetricBuildItem.java

```diff
@@ -0,0 +1,75 @@
+package io.quarkus.smallrye.metrics.deployment.spi;
+
+import org.eclipse.microprofile.metrics.Metadata;
+import org.eclipse.microprofile.metrics.MetricType;
+import org.eclipse.microprofile.metrics.Tag;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * Build item that is picked up by the SmallRye Metrics extension to register metrics required by other extensions.
+ */
+public final class MetricBuildItem extends MultiBuildItem {
+
+    private final Metadata metadata;
+    private final Tag[] tags;
+    private final Object implementor;
+    private final boolean enabled;
+
+    /**
+     * Create a metric build item from the specified metadata and tags.
+     * Such metric will be picked up by the Metrics extension and registered in the VENDOR registry.
+     * This constructor is applicable to all metric types except gauges.
+     *
+     * @param metadata The metadata that should be applied to the registered metric
+     * @param enabled Whether this metric is enabled
+     * @param tags The tags that will be applied to this metric
+     */
+    public MetricBuildItem(Metadata metadata, boolean enabled, Tag... tags) {
```

Comment:
You should also add the `configRoot` field to your MetricBuildItem as it is done to Health.
The configRoot is not the same as the extension name (for agroal, the config root is datasource).

This is for a future usage to implements more complex enabling/disabling of extension metrics like 
```
quarkus.metrics.extensions=transaction,kafka
# enabled metrics for the transaction and kafka extension only
```

Reply:
Done